tests/unit/amo/components/TestAddonQRCodeLink.js

```diff
@@ -0,0 +1,159 @@
+        _config,
+        addon: goodAddon,
+      });
+
+      expect(root.find('.AddonQRCodeLink')).toHaveLength(1);
+    });
+
+    it('does not display a link for a QR code on desktop, for an invalid add-on', () => {
+      dispatchClientMetadata({ clientApp: CLIENT_APP_FIREFOX, store });
+      const root = render({
+        _config,
+        addon: createInternalAddonWithLang({
+          ...fakeAddon,
+          id: goodAddonId + 1,
+        }),
+      });
+
+      expect(root.find('.AddonQRCodeLink')).toHaveLength(0);
+    });
+
+    it('does not display a link for a QR code on mobile, for an applicable add-on', () => {
+      dispatchClientMetadata({ clientApp: CLIENT_APP_ANDROID, store });
+      const root = render({
+        _config,
+        addon: goodAddon,
+      });
+
+      expect(root.find('.AddonQRCodeLink')).toHaveLength(0);
+    });
+
+    it('displays a modal when user clicks the QRCode link', () => {
+      const preventDefaultSpy = sinon.spy();
+      dispatchClientMetadata({ clientApp: CLIENT_APP_FIREFOX, store });
+      const root = render({
+        _config,
+        addon: goodAddon,
+      });
+
+      expect(root.find('.AddonQRCodeLink-modal')).toHaveLength(0);
+
+      root
+        .find('.AddonQRCodeLink-button')
+        .simulate(
+          'click',
+          createFakeEvent({ preventDefault: preventDefaultSpy }),
+        );
+
+      sinon.assert.called(preventDefaultSpy);
+
+      const modal = root.find('.AddonQRCodeLink-modal');
+
+      expect(modal).toHaveLength(1);
+      expect(modal).toHaveProp('id', 'AddonQRCodeLink-modal');
+
+      expect(root.find(AddonQRCode)).toHaveLength(1);
+      expect(root.find(AddonQRCode)).toHaveProp('addon', goodAddon);
+      expect(root.find(AddonQRCode)).toHaveProp(
+        'onDismiss',
+        root.instance().onHideQRCode,
+      );
+    });
+
+    it('closes the modal when user clicks the dismiss button', () => {
+      const preventDefaultSpy = sinon.spy();
+      dispatchClientMetadata({ clientApp: CLIENT_APP_FIREFOX, store });
+      const root = render({
+        _config,
+        addon: goodAddon,
+      });
+
+      expect(root.find('.AddonQRCodeLink-modal')).toHaveLength(0);
+
+      root
+        .find('.AddonQRCodeLink-button')
+        .simulate(
+          'click',
+          createFakeEvent({ preventDefault: preventDefaultSpy }),
+        );
+
+      sinon.assert.called(preventDefaultSpy);
+
+      expect(root.find('.AddonQRCodeLink-modal')).toHaveLength(1);
+
+      // Simulate clicking on the dismiss button.
+      root.instance().onHideQRCode();
+
+      expect(root.find('.AddonQRCodeLink-modal')).toHaveLength(0);
+    });
+  });
+
+  describe('flag disabled', () => {
+    const _config = getFakeConfig({
+      ...testConfig,
+      enableFeatureAddonQRCode: false,
+    });
+
+    it('does not display a link for a QR code on desktop, for an applicable add-on', () => {
+      dispatchClientMetadata({ clientApp: CLIENT_APP_FIREFOX, store });
```

Comment:
In all the test cases, there is a call to `dispatchClientMetadata()` with the global-ish `store`. I don't quite like the idea of having a shared `store` even if it gets reset before each test.

In many other test files, we are more conservative. We pass the `store` to `render()` (which has a default store otherwise): https://github.com/mozilla/addons-frontend/blob/a75225317cc58768876d969bce93f1ba49f53555/tests/unit/amo/pages/TestBlock.js#L23

I'd like to do that in this test file as well. It shouldn't change the number of LOC but it should remove this global-ish variable:

```js
const { store } = dispatchClientMetadata({ clientApp: CLIENT_APP_FIREFOX });
// ...

const root = render({ config, store, addon: goodAddon });

// ...
```

Reply:
While it's true that we create the `store` in the test, and then pass it to `render` in many test cases, the pattern of having a global `store` is also common. I see it in 38 other files, so it is indeed quite common. Just doing a rudimentary search for the other pattern (the one you suggest) shows 47 test files, so they actually seem to be almost equally common.

I am fine with changing this here, but I wonder, if this is going to be something called out in a code review, whether we should have some sort of style guide that explains that this is the standard. As this is quite a common practice, it isn't something that I would normally call out in a code review.